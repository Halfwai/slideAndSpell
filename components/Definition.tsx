import { View, Text, StyleSheet } from 'react-native';

interface DefinitionProps {
    word: string;
    definition: string;
}


export default function Definition(props: DefinitionProps) {
    console.log(props.word, props.definition);
    return (
        <View style={styles.container}>
            <Text>{props.word}</Text>
            <Text>{props.definition}</Text>
        </View>
    )
} 

const styles = StyleSheet.create({
    container: {

        alignItems: 'center',
        width: "90%",
    }
});