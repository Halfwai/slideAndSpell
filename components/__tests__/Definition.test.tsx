import { render } from '@testing-library/react-native';

import Definition from '@/components/submenuComponents/Definition';

it(`Definition Component renders correctly`, () => {
    const { toJSON } = render(
        <Definition word={"test"} definition={"a test definition"} example={"test example"} />
    );
    expect(toJSON()).toMatchSnapshot();
});