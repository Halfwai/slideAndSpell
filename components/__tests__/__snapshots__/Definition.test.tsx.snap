// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Definition Component renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "borderColor": "#8BF0BC",
      "borderRadius": 10,
      "borderWidth": 1,
      "margin": 10,
      "padding": 20,
      "width": 675,
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Quicksand",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          {
            "fontSize": 18,
            "fontStyle": "italic",
            "textTransform": "capitalize",
          },
        ]
      }
    >
      test
      : 
    </Text>
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          {
            "fontSize": 18,
          },
        ]
      }
    >
      a test definition
    </Text>
  </Text>
  <Text
    style={
      [
        {
          "fontFamily": "Quicksand",
        },
        {
          "fontSize": 16,
          "fontStyle": "italic",
        },
      ]
    }
  >
    "
    test example
    "
  </Text>
  <Text
    href="/dictionary?word=test"
    onPress={[Function]}
    role="link"
    style={
      {
        "alignSelf": "flex-end",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginTop": 10,
        "padding": 5,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Quicksand",
          },
          undefined,
        ]
      }
    >
      More info
    </Text>
  </Text>
</View>
`;
